{"links": {"self": {"href": "data/repositories/osrf/drcsim/issues/434/comments/15607894.json"}, "html": {"href": "#!/osrf/drcsim/issues/434#comment-15607894"}}, "issue": {"links": {"self": {"href": "data/repositories/osrf/drcsim/issues/434.json"}}, "type": "issue", "id": 434, "repository": {"links": {"self": {"href": "data/repositories/osrf/drcsim.json"}, "html": {"href": "#!/osrf/drcsim"}, "avatar": {"href": "data/bytebucket.org/ravatar/{247b5408-ae91-4e3c-b2f9-a21994658d11}ts=c_plus_plus"}}, "type": "repository", "name": "drcsim", "full_name": "osrf/drcsim", "uuid": "{247b5408-ae91-4e3c-b2f9-a21994658d11}"}, "title": "Boost mutex error"}, "content": {"raw": "Looks like a shutdown issue, proper environment path setup avoids this error. Is this blocking?\n\nLooking at the original bug report, it seems the failure is triggered by not being able to find the world file, subsequently some component failed to shutdown properly and producing the error.\n\n~~~\n$ roslaunch drcsim_gazebo atlas.launch extra_gazebo_args:=\"--verbose\" gzname:=\"gzserver_gdb\"\n...\n[Err] [Server.cc:285] Could not open file[atlas.world]\n...\nProgram received signal SIGSEGV, Segmentation fault.\n[Switching to Thread 0x7fffb67fc700 (LWP 20489)]\n__memcpy_sse2_unaligned ()\n    at ../sysdeps/x86_64/multiarch/memcpy-sse2-unaligned.S:36\n36\t../sysdeps/x86_64/multiarch/memcpy-sse2-unaligned.S: No such file or directory.\n(gdb) bt\n#0  __memcpy_sse2_unaligned ()\n    at ../sysdeps/x86_64/multiarch/memcpy-sse2-unaligned.S:36\n#1  0x00007ffff537ffa0 in std::string::_Rep::_M_clone(std::allocator<char> const&, unsigned long) () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6\n#2  0x00007ffff53805fc in std::basic_string<char, std::char_traits<char>, std::allocator<char> >::basic_string(std::string const&) ()\n   from /usr/lib/x86_64-linux-gnu/libstdc++.so.6\n#3  0x00007fffd9419d64 in ros::param::getImpl(std::string const&, XmlRpc::XmlRpcValue&, bool) () from /opt/ros/indigo/lib/libroscpp.so\n#4  0x00007fffd941ad59 in ros::param::getImpl(std::string const&, double&, bool) () from /opt/ros/indigo/lib/libroscpp.so\n#5  0x00007fffd945bf9c in ros::NodeHandle::getParam(std::string const&, double&) const () from /opt/ros/indigo/lib/libroscpp.so\n#6  0x00007fffd99ac67b in gazebo_ros::PhysicsConfig::__fromServer__(ros::NodeHandle const&) ()\n   from /home/hsu/projects/catkin_drcsim_drcsim4_release/install/lib/libgazebo_ros_api_plugin.so\n#7  0x00007fffd99ae04f in dynamic_reconfigure::Server<gazebo_ros::PhysicsConfig>::init() ()\n   from /home/hsu/projects/catkin_drcsim_drcsim4_release/install/lib/libgazebo_ros_api_plugin.so\n#8  0x00007fffd9973b40 in gazebo::GazeboRosApiPlugin::physicsReconfigureThread() ()\n   from /home/hsu/projects/catkin_drcsim_drcsim4_release/install/lib/libgazebo_ros_api_plugin.so\n#9  0x00007ffff5c5ca4a in ?? ()\n   from /usr/lib/x86_64-linux-gnu/libboost_thread.so.1.54.0\n#10 0x00007ffff6a9e182 in start_thread (arg=0x7fffb67fc700)\n    at pthread_create.c:312\n#11 0x00007ffff4ade00d in clone ()\n    at ../sysdeps/unix/sysv/linux/x86_64/clone.S:111\n(gdb) \n~~~\n\nreiterating the error:\n~~~\n$ roslaunch drcsim_gazebo atlas.launch extra_gazebo_args:=\"--verbose\" \n...\n[Err] [Server.cc:285] Could not open file[atlas.world]\nterminate called after throwing an instance of 'boost::exception_detail::clone_impl<boost::exception_detail::error_info_injector<boost::lock_error> >'\n  what():  boost: mutex unlock failed in pthread_mutex_unlock: Invalid argument\n[gazebo-1] process has died [pid 21036, exit code 255, cmd /home/hsu/projects/catkin_drcsim_drcsim4_release/install/share/drcsim_gazebo/scripts/run_gazebo atlas.world --verbose __name:=gazebo __log:=/home/hsu/.ros/log/bc7187c6-b173-11e4-8d3b-d0509919a838/gazebo-1.log].\nlog file: /home/hsu/.ros/log/bc7187c6-b173-11e4-8d3b-d0509919a838/gazebo-1*.log\n~~~", "markup": "markdown", "html": "<p>Looks like a shutdown issue, proper environment path setup avoids this error. Is this blocking?</p>\n<p>Looking at the original bug report, it seems the failure is triggered by not being able to find the world file, subsequently some component failed to shutdown properly and producing the error.</p>\n<div class=\"codehilite\"><pre><span></span>$ roslaunch drcsim_gazebo atlas.launch extra_gazebo_args:<span class=\"o\">=</span><span class=\"s2\">&quot;--verbose&quot;</span> gzname:<span class=\"o\">=</span><span class=\"s2\">&quot;gzserver_gdb&quot;</span>\n...\n<span class=\"o\">[</span>Err<span class=\"o\">]</span> <span class=\"o\">[</span>Server.cc:285<span class=\"o\">]</span> Could not open file<span class=\"o\">[</span>atlas.world<span class=\"o\">]</span>\n...\nProgram received signal SIGSEGV, Segmentation fault.\n<span class=\"o\">[</span>Switching to Thread 0x7fffb67fc700 <span class=\"o\">(</span>LWP <span class=\"m\">20489</span><span class=\"o\">)]</span>\n__memcpy_sse2_unaligned <span class=\"o\">()</span>\n    at ../sysdeps/x86_64/multiarch/memcpy-sse2-unaligned.S:36\n<span class=\"m\">36</span>  ../sysdeps/x86_64/multiarch/memcpy-sse2-unaligned.S: No such file or directory.\n<span class=\"o\">(</span>gdb<span class=\"o\">)</span> bt\n<span class=\"c1\">#0  __memcpy_sse2_unaligned ()</span>\n    at ../sysdeps/x86_64/multiarch/memcpy-sse2-unaligned.S:36\n<span class=\"c1\">#1  0x00007ffff537ffa0 in std::string::_Rep::_M_clone(std::allocator&lt;char&gt; const&amp;, unsigned long) () from /usr/lib/x86_64-linux-gnu/libstdc++.so.6</span>\n<span class=\"c1\">#2  0x00007ffff53805fc in std::basic_string&lt;char, std::char_traits&lt;char&gt;, std::allocator&lt;char&gt; &gt;::basic_string(std::string const&amp;) ()</span>\n   from /usr/lib/x86_64-linux-gnu/libstdc++.so.6\n<span class=\"c1\">#3  0x00007fffd9419d64 in ros::param::getImpl(std::string const&amp;, XmlRpc::XmlRpcValue&amp;, bool) () from /opt/ros/indigo/lib/libroscpp.so</span>\n<span class=\"c1\">#4  0x00007fffd941ad59 in ros::param::getImpl(std::string const&amp;, double&amp;, bool) () from /opt/ros/indigo/lib/libroscpp.so</span>\n<span class=\"c1\">#5  0x00007fffd945bf9c in ros::NodeHandle::getParam(std::string const&amp;, double&amp;) const () from /opt/ros/indigo/lib/libroscpp.so</span>\n<span class=\"c1\">#6  0x00007fffd99ac67b in gazebo_ros::PhysicsConfig::__fromServer__(ros::NodeHandle const&amp;) ()</span>\n   from /home/hsu/projects/catkin_drcsim_drcsim4_release/install/lib/libgazebo_ros_api_plugin.so\n<span class=\"c1\">#7  0x00007fffd99ae04f in dynamic_reconfigure::Server&lt;gazebo_ros::PhysicsConfig&gt;::init() ()</span>\n   from /home/hsu/projects/catkin_drcsim_drcsim4_release/install/lib/libgazebo_ros_api_plugin.so\n<span class=\"c1\">#8  0x00007fffd9973b40 in gazebo::GazeboRosApiPlugin::physicsReconfigureThread() ()</span>\n   from /home/hsu/projects/catkin_drcsim_drcsim4_release/install/lib/libgazebo_ros_api_plugin.so\n<span class=\"c1\">#9  0x00007ffff5c5ca4a in ?? ()</span>\n   from /usr/lib/x86_64-linux-gnu/libboost_thread.so.1.54.0\n<span class=\"c1\">#10 0x00007ffff6a9e182 in start_thread (arg=0x7fffb67fc700)</span>\n    at pthread_create.c:312\n<span class=\"c1\">#11 0x00007ffff4ade00d in clone ()</span>\n    at ../sysdeps/unix/sysv/linux/x86_64/clone.S:111\n<span class=\"o\">(</span>gdb<span class=\"o\">)</span> \n</pre></div>\n\n\n<p>reiterating the error:</p>\n<div class=\"codehilite\"><pre><span></span>$ roslaunch drcsim_gazebo atlas.launch extra_gazebo_args:<span class=\"o\">=</span><span class=\"s2\">&quot;--verbose&quot;</span> \n...\n<span class=\"o\">[</span>Err<span class=\"o\">]</span> <span class=\"o\">[</span>Server.cc:285<span class=\"o\">]</span> Could not open file<span class=\"o\">[</span>atlas.world<span class=\"o\">]</span>\nterminate called after throwing an instance of <span class=\"s1\">&#39;boost::exception_detail::clone_impl&lt;boost::exception_detail::error_info_injector&lt;boost::lock_error&gt; &gt;&#39;</span>\n  what<span class=\"o\">()</span>:  boost: mutex unlock failed in pthread_mutex_unlock: Invalid argument\n<span class=\"o\">[</span>gazebo-1<span class=\"o\">]</span> process has died <span class=\"o\">[</span>pid <span class=\"m\">21036</span>, <span class=\"nb\">exit</span> code <span class=\"m\">255</span>, cmd /home/hsu/projects/catkin_drcsim_drcsim4_release/install/share/drcsim_gazebo/scripts/run_gazebo atlas.world --verbose __name:<span class=\"o\">=</span>gazebo __log:<span class=\"o\">=</span>/home/hsu/.ros/log/bc7187c6-b173-11e4-8d3b-d0509919a838/gazebo-1.log<span class=\"o\">]</span>.\nlog file: /home/hsu/.ros/log/bc7187c6-b173-11e4-8d3b-d0509919a838/gazebo-1*.log\n</pre></div>", "type": "rendered"}, "created_on": "2015-02-10T22:37:22.199893+00:00", "user": {"display_name": "John Hsu", "uuid": "{0a186eae-abf0-4514-a951-23db5eccc286}", "links": {"self": {"href": "https://api.bitbucket.org/2.0/users/%7B0a186eae-abf0-4514-a951-23db5eccc286%7D"}, "html": {"href": "https://bitbucket.org/%7B0a186eae-abf0-4514-a951-23db5eccc286%7D/"}, "avatar": {"href": "https://avatar-management--avatars.us-west-2.prod.public.atl-paas.net/557058:f3968cd3-4910-4384-8349-482a6c7889ec/5445ce6e-6273-47f0-84eb-621c86ca11cb/128"}}, "nickname": "hsu", "type": "user", "account_id": "557058:f3968cd3-4910-4384-8349-482a6c7889ec"}, "updated_on": "2015-02-10T22:39:31.751456+00:00", "type": "issue_comment", "id": 15607894}